v2/pkg/clusterresources/interface.go

```diff
@@ -0,0 +1,12 @@
+package clusterresources
+
+import (
+	"context"
+
+	"github.com/openshift/oc-mirror/v2/pkg/api/v1alpha3"
+	"github.com/openshift/oc-mirror/v2/pkg/mirror"
+)
+
+type GeneratorInterface interface {
```

Comment:
Having only one interface for the generation of all cluster resources could impact in future developments when there is a need to have different implementations of IDMS and ITMS.

Maybe having a separate interface for each resource (for example one for IDMS and another for ITMS) could be a good idea to make it even more flexible.

Using interface is powerful, thanks for adding it @sherine-k

Reply:
Yes absolutely @aguidirh!
I was thinking of adding another method in there... ITMSGenerator...
There might also be a need later to create IDMS and ITMS files per category of mirrored images (release-idms, operator-idms, etc).
So these method signatures will surely need to evolve as we collect our requirements better.
I'm happy to adapt this according to what we have so far.